sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchAsyncClient.java

```diff
@@ -475,20 +476,29 @@ SearchPagedFlux search(String searchText, SearchOptions searchOptions, RequestOp
      * @see <a href="https://docs.microsoft.com/rest/api/searchservice/Lookup-Document">Lookup document</a>
      */
     @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<Response<SearchDocument>> getDocumentWithResponse(String key, List<String> selectedFields,
-        RequestOptions requestOptions) {
-        return withContext(context -> getDocumentWithResponse(key, selectedFields, requestOptions, context));
+    public <T> Mono<Response<T>> getDocumentWithResponse(String key, Class<T> modelClass,
+        List<String> selectedFields, RequestOptions requestOptions) {
+        return withContext(context -> getDocumentWithResponse(key, modelClass,
+            selectedFields, requestOptions, context));
     }
 
-    Mono<Response<SearchDocument>> getDocumentWithResponse(String key, List<String> selectedFields,
-        RequestOptions requestOptions, Context context) {
+    @SuppressWarnings("unchecked")
+    <T> Mono<Response<T>> getDocumentWithResponse(String key, Class<T> modelClass,
+        List<String> selectedFields, RequestOptions requestOptions, Context context) {
         try {
 
             return restClient.documents()
                 .getWithRestResponseAsync(key, selectedFields, RequestOptionsConverter.map(requestOptions), context)
                 .onErrorMap(DocumentResponseConversions::exceptionMapper)
                 .map(res -> {
-                    SearchDocument document = MAPPER.convertValue(res.getValue(), SearchDocument.class);
+                    SearchObjectMapper.getInstance().setSerializationInclusion(JsonInclude.Include.ALWAYS);
```

Comment:
Any reason we aren't setting the `setSerializationInclusion` property when we are constructing the mapper instance? This leads to the change where the static object is being mutated while being used, possibly running into race conditions.

Reply:
Good catch. This can be put in property when construct.